### PR TITLE
Blog-app : add devise for sign up user

### PR DESCRIPTION
**Pull request by tajulafreen**
**In this PR I did the following:**

- Removed current_user method in ApplicationController, devise will provide with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password is stored in the database.
- Asked for confirmation of email.
- Can reset password.

### CODE REVIEW

**Comment by algerina** [comment]
Hi @tajulafreen , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project where Linters are not passing.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @tajulafreen 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Invalid Code Review Request 🔴 

You have submitted a project, that does not have your partner's commit. ensure that he has contributed equally to the project  before submitting another review

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (cynthiainga) [inline app/models/user.rb]: - [ ] As mentioned in the project requirement `:confirmable` should be enabled in the user mode, user should be asked for confirmation of email. Kindly add it 🙃 🙃

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 class UsersController < ApplicationController
+  before_action :authenticate_user!
   def index
     @users = User.all
+    # @user = current_user
```
Comment (cynthiainga) [inline app/controllers/users_controller.rb]: - [ ] Please kindly delete the commented line of code to keep your code clean and look more professional 😁

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @tajulafreen, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] The `current_user` method in `ApplicationController` has been removed ✔️ 
- [ ] Devise has been installed ✔️ 
- [ ] The application allows for registering a new user ✔️ 
- [ ] A user can log in with a combination of email and password ✔️ 
- [ ] A user can reset the password in the application ✔️ 
- [ ] A hashed password is stored in the database ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @tajulafreen, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 
: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**